ceh-2000/gpu_guru
# WIP: adding non-IID dataset splitting

Re-implementing sampling from a Dirichlet distribution to simulate non-IID label splits for classification tasks.

Still need to:
- [x] Implement non-IID splitting function and integrate into framework
- [x] Do some basic testing (first with `sample_ratio=1`, then with `sample_ratio<1`)
- [x] Control all random elements with seeds for consistency
- [x] Add `seaborn` visualization to verify the heterogeneity of the dataset split
- [x] Apply `black` and `isort` for formatting

Testing:
1. Pull the branch
2. Run `python3 main.py --trials 1 --dataset mnist --num_users 10 --glob_epochs 3 --local_epochs 1 --should_log True`, which should produce the same result as previous code
3. Run `python3 main.py --trials 1 --dataset mnist --num_users 10 --glob_epochs 3 --local_epochs 1 --should_log True --alpha 0.1 --sample_ratio 0.5` to split in a non-IID manner (inspect graph that is displayed to confirm!)
